src/crimson/common/errorator.h

```diff
@@ -66,11 +68,8 @@ inline auto do_until(AsyncAction action) {
           return ::crimson::do_until(
 	    std::move(action));
```

Comment:
Looks `crimson::do_until()` has further issues under **non-debug** build.

My understanding is that here it transforms iterative calls to recursive calls and will crush the program if the iteration continues.

The observation is that when the related seg fault happens, the backtrace will contain 80+ recursive calls at the current line.

Is there a proper way to transform back to iterative calls if the `action` is synchronous?

Reply:
this is not true. after this change, if the state of the return future is available and there is no pending continuation in the queue of the reactor, the `while (true)` loop starts another iteration.

the 80+ recursive calls is called by the buggy implementation, where the state of `f` is reset by `f.get()` is called before the `if (!f.available() || seastar::need_preempt()) {...}` block.